processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -133,13 +173,20 @@ protected AddToFactsResult addToFacts(
 
       // Last ditch sanity checks
       if (numEntries.get() >= maxRowCount
-          && facts.getPriorIndex(key) == IncrementalIndexRow.EMPTY_ROW_INDEX
-          && !skipMaxRowsInMemoryCheck) {
-        throw new IndexSizeExceededException("Maximum number of rows [%d] reached", maxRowCount);
+          || (maxBytesInMemory != -1 && sizeInBytes.get() >= maxBytesInMemory)
```

Comment:
`maxBytesInMemory > 0` would be better (https://github.com/druid-io/druid/pull/5583#discussion_r179689109).

Reply:
Probably `(numEntries.get() >= maxRowCount || maxBytesInMemory > 0 && sizeInBytes.get() >= maxBytesInMemory)` to check to meet one of either conditions?